.github/workflows/markdown-linter.yml

```diff
@@ -0,0 +1,35 @@
+name: "Markdown Lint and Link Check"
```

Comment:
[optional][uber nit] In the past I've had a difficult time mapping from Action name in the sidebar here

<img width="349" alt="Screenshot 2023-09-19 at 12 08 16 PM" src="https://github.com/celestiaorg/celestia-app/assets/3699047/73fcd9a9-0ced-4b8c-ac81-9bb71152536c">

to the file that defined the action. I think we could make it easier to map back and forth by calling the action and the file the same thing

```suggestion
name: "markdown-linter"
```